plugins/ipfs/local/localipfs.go

```diff
@@ -0,0 +1,504 @@
+		return nil, err
+	}
+
+	lt := ipfs.LT_TCP
+	mdns := false
+
+	if v, ok := attrs["listentype"]; ok {
+		ltstr, ok := v.(string)
+
+		if !ok {
+			return nil, fmt.Errorf("Attr `listentype` should be a string")
+		}
+
+		switch ltstr {
+		case "":
+			lt = ipfs.LT_TCP
+			break
+		case "ws":
+			lt = ipfs.LT_WS
+			break
+		case "utp":
+			lt = ipfs.LT_UTP
+			break
+		default:
+			return nil, fmt.Errorf("Unsupported `listentype` %s", ltstr)
+		}
+	}
+
+	if _, ok := attrs["mdns"]; ok {
+		mdns = true
+	}
+
+	return &Localipfs{
+		dir:  dir,
+		lt:   lt,
+		mdns: mdns,
+	}, nil
+
+}
+
+func GetAttrList() []string {
+	return ipfs.GetAttrList()
+}
+
+func GetAttrDesc(attr string) (string, error) {
+	return ipfs.GetAttrDesc(attr)
+}
+
+func GetMetricList() []string {
+	return ipfs.GetMetricList()
+}
+
+func GetMetricDesc(attr string) (string, error) {
+	return ipfs.GetMetricDesc(attr)
+}
+
+/// TestbedNode Interface
+
+func (l *Localipfs) Init(ctx context.Context, agrs ...string) (testbedi.Output, error) {
+	agrs = append([]string{"ipfs", "init"}, agrs...)
+	output, oerr := l.RunCmd(ctx, nil, agrs...)
+	if oerr != nil {
+		return nil, oerr
+	}
+
+	icfg, err := l.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	lcfg := icfg.(*config.Config)
+
+	lcfg.Bootstrap = nil
+	lcfg.Addresses.Swarm = []string{ipfs.SwarmAddr(l.lt, "127.0.0.1", 0)}
+	lcfg.Addresses.API = fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", 0)
+	lcfg.Addresses.Gateway = ""
+	lcfg.Discovery.MDNS.Enabled = l.mdns
+
+	err = l.WriteConfig(lcfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return output, oerr
+}
+
+func (l *Localipfs) Start(ctx context.Context, wait bool, args ...string) (testbedi.Output, error) {
+	alive, err := l.isAlive()
+	if err != nil {
+		return nil, err
+	}
+
+	if alive {
+		return nil, fmt.Errorf("node is already running")
+	}
+
+	dir := l.dir
+	dargs := append([]string{"daemon"}, args...)
+	cmd := exec.Command("ipfs", dargs...)
+	cmd.Dir = dir
+
+	cmd.Env, err = l.env()
+	if err != nil {
+		return nil, err
+	}
+
+	iptbutil.SetupOpt(cmd)
+
+	stdout, err := os.Create(filepath.Join(dir, "daemon.stdout"))
+	if err != nil {
+		return nil, err
+	}
+
+	stderr, err := os.Create(filepath.Join(dir, "daemon.stderr"))
+	if err != nil {
+		return nil, err
+	}
+
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	err = cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	pid := cmd.Process.Pid
+
+	err = ioutil.WriteFile(filepath.Join(dir, "daemon.pid"), []byte(fmt.Sprint(pid)), 0666)
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, ipfs.WaitOnAPI(l)
+}
+
+func (l *Localipfs) Stop(ctx context.Context, wait bool) error {
+	pid, err := l.getPID()
+	if err != nil {
+		return fmt.Errorf("error killing daemon %s: %s", l.dir, err)
+	}
+
+	p, err := os.FindProcess(pid)
+	if err != nil {
+		return fmt.Errorf("error killing daemon %s: %s", l.dir, err)
+	}
+
+	waitch := make(chan struct{}, 1)
+	go func() {
+		p.Wait() //TODO: pass return state
```

Comment:
should this be fixed before merging?

Reply:
I believe this was copied over from some other code. Not sure what the original intent was. I've removed the todo.